api/utils/iter/sorted.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iter
+
+import "github.com/gravitational/trace"
+
+// joinSorted joins two sorted iterators into a single sorted iterator.
```

Comment:
What kind of entries are we sorting here? Is it something where we can use a comparable constraint and operator `<` instead of a function call for each comparison?

Reply:
Moved to [fspmarshall/sorted-stream-helpers](https://github.com/gravitational/teleport/tree/fspmarshall/sorted-stream-helpers) for now.